### PR TITLE
Added metrics: cache size, serialize time, set snapshot time

### PR DESCRIPTION
**Pull request by lukidzi**

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt [hunk 1]
```diff
@@ -138,7 +139,9 @@ class SnapshotUpdater(
     fun updateSnapshotForGroup(group: Group, globalSnapshot: Snapshot) {
         try {
             val groupSnapshot = snapshotFactory.getSnapshotForGroup(group, globalSnapshot)
+            val setSnapshot = Timer.start(meterRegistry)
             cache.setSnapshot(group, groupSnapshot)
+            setSnapshot.stop(meterRegistry.timer("snapshot-updater.set-snapshot.${group.serviceName}.time"))
```
Comment (pbetkier) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt]:
```
meterRegistry.timer("...").record(
    () -> cache.setSnapshot(group, groupSnapshot)
)
```
would be more readable to me.

Comment (pbetkier) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt]: same in serializer class

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/server/CachedProtoResourcesSerializer.kt [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
-    private val cache = CacheBuilder.newBuilder()
-        .weakValues()
-        .build<Collection<Message>, MutableCollection<Any>>()
+    private val monitorCache: Cache<Collection<Message>, MutableCollection<Any>> = GuavaCacheMetrics
+        .monitor(meterRegistry,
+            CacheBuilder.newBuilder()
+                .recordStats()
+                .weakValues()
+                .build<Collection<Message>, MutableCollection<Any>>(),
+            "protoCache", "cacheName", "stat")
```
Comment (MarcinFalkowski) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/server/CachedProtoResourcesSerializer.kt]: what is "cacheName" and "stat"? I suspect we can choose more meaningful names.

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/server/CachedProtoResourcesSerializer.kt [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
 
     override fun serialize(resources: MutableCollection<out Message>): MutableCollection<Any> {
-        return cache.get(resources) {
-            resources.asSequence()
-                .map { Any.pack(it) }
-                .toMutableList()
+        return meterRegistry.timer("proto-cache.serialize.${resources.size}.time").recordCallable{
```
Comment (MarcinFalkowski) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/server/CachedProtoResourcesSerializer.kt]: Can we use `.record {}` like in other class?

**Pull request merged**